h/session.py

```diff
@@ -47,21 +48,43 @@ def _current_groups(request):
     This list is meant to be returned to the client in the "session" model.
 
     """
-    groups = [
-        {'name': 'Public', 'id': '__world__', 'public': True},
-    ]
+
     user = request.authenticated_user
+    authority = user.authority if user else request.auth_domain
+
+    groups = (_authority_groups(request.auth_domain, authority) +
+              _user_groups(user))
+
+    return [_group_model(request.route_url, group) for group in groups]
+
+
+def _authority_groups(auth_domain, authority):
+    """Return the default groups associated with an authority."""
+    if authority == auth_domain:
+        return [util.WorldGroup(auth_domain)]
+    else:
+        return []
```

Comment:
For a logged-in third party user, this currently results in an empty list of groups being returned to the client because the user is not an explicit member of a publisher group (as returned by `user.groups`).

See [Slack discussion](https://hypothes-is.slack.com/archives/public/p1485943172003668)